Refactor and document proxy mode. (#65)
MangoDB-io/MangoDB: CONTRIBUTING.md

MangoDB development is done on the host running Linux or macOS, with everything else running inside Docker Compose.

> You would need Go 1.18 that is not released yet.

Reviewer: ```suggestion
You will need Go 1.18 that is not released yet.
```

[Compile it yourself](https://golang.org/doc/install/source) or use [`gotip download`](https://pkg.go.dev/golang.org/dl/gotip).
Verify Go version:
```
$ go version
go version devel go1.18-[...]
```

1. Install tools with `make init`.
2. Start the development environment with `make env-up`.

>    That would start PostgreSQL and MongoDB and fill them with identical sets of testing data.

Reviewer: ```suggestion
   This will start PostgreSQL and MongoDB and fill them with identical sets of testing data.
```

3. Run tests in the other window with `make test`.
4. See all available targets with `make help`.
5. Start MangoDB with `make run`.

>    That would start ot in development mode when all requests are both handled by MangoDB and routed to MongoDB.

Reviewer: ```suggestion
   That will start it in a development mode where all requests are handled by MangoDB and also routed to MongoDB.
```
